From 9b713a06ad123e8d4bba7c08421511b238da2e10 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marek=20Marczykowski-G=C3=B3recki?=
 <marmarek@invisiblethingslab.com>
Date: Mon, 1 Feb 2021 05:40:27 +0100
Subject: [PATCH] Allow empty reason for dbus Inhibit method

Specification doesn't explicitly require it and some applications (like
xfce4-power-manager) use empty one.
---
 driver/xscreensaver-systemd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/driver/xscreensaver-systemd.c b/driver/xscreensaver-systemd.c
index 2da99de..b3676ec 100644
--- a/driver/xscreensaver-systemd.c
+++ b/driver/xscreensaver-systemd.c
@@ -526,7 +526,7 @@ xscreensaver_method_inhibit (sd_bus_message *m, void *arg,
     return -1;
   }
 
-  if (!inhibit_reason || !*inhibit_reason) {
+  if (!inhibit_reason) {
     fprintf (stderr, "%s: no reason in method call from \"%s\"\n",
              blurb(), application_name);
     return -1;
-- 
2.26.2

